CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -76,4 +77,34 @@ private static function validateDates(array $params) {
     }
   }
 
+
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array $params
+   *  Request Parameters
+   *
+   * @throws \Exception
+   */
+  private static function validateReviewFields(array $params) {
+    $reviewFieldsLegalOptions = ['id', 'weight', 'required'];
+    $reviewFieldsOptions = CRM_Utils_Array::value('review_fields', $params);
+    if (!$reviewFieldsOptions) {
+      throw new Exception(ts('Valid review fields options must be set'));
+    }
+
+    foreach ($reviewFieldsOptions as $reviewFieldsOption) {
+      $missingFieldsDiff = array_diff($reviewFieldsLegalOptions, array_keys($reviewFieldsOption));
```

Comment:
Maybe we can go a little step further and check that the values of these fields are what we expect.
We can restructure the reviewFieldsLegalOption array and use the https://www.php.net/manual/en/function.filter-var.php function to do the validation
```
  $reviewFieldsLegalOptions = [
    [
      'name' => 'id',
      'validation' => FILTER_VALIDATE_INT
    ],
    
    [
    'name' => 'weight',
    'validation' => FILTER_VALIDATE_INT
    ]
  ];
```